### ISSUE TITLE
Unserialize function usage

### ISSUE THREAD

**Issue by volodymyrdovhan** [WisdmLabs/moodle-auth_edwiserbridge#26]
Hello,
unserialize() function is defined as a malicious function.
https://moodledev.io/general/community/plugincontribution/checklist#security
It should be replaced with some more secure solution, like json_encode/json_decode or something else.

![image](https://github.com/user-attachments/assets/a83b283b-2a88-4547-97f3-33e2e6118a7b)

**Comment by jaydeeptrivedi13** [comment]
@volodymyrdovhan I agree with the issue. However, we can't directly remove all instances of unserialize as the plugin is installed on many client sites. 
**For now,  we have implemented a migration script which will convert any serialised data to json and store that in db. Also, all instances of fetch and set have been converted to json_encode/decode. The only instances you will find are in the migration script and in the licensing logic where we have added additional handling just in case the data is returned in the serialised format.**
